internal/v4/auth.go

```diff
@@ -61,14 +72,61 @@ func (h *Handler) authenticate(w http.ResponseWriter, req *http.Request, id *cha
 	return httpbakery.NewDischargeRequiredError(m, verr)
 }
 
+func (h *Handler) authorizeEntity(id *charm.Reference, req *http.Request) error {
+	// TODO frankban: the first time a new charm is published, its
+	// corresponding base entity is not yet present in the database.
+	// For this reason, the check below must still allow specific users to
```

Comment:
This doesn't sound like a TODO to me. I think this will continue to be relevant indefinitely.